js/boaexchange.js

```diff
@@ -0,0 +1,1163 @@
+        return this.extend (this.parseOrder (response['data'], market), {
+            'status': 'open',
+            'price': order['price'],
+            'symbol': symbol,
+            'amount': order['amount'],
+            'side': side,
+            'type': type,
+            'id': response['data']['id'],
+        });
+    }
+
+    async createLimitOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        let response = await this.createOrder (symbol, type, side, price, params);
+        return response;
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let request = { 'orderId': id };
+        let response = await this.v1DeleteOrdersOrderId (this.extend (request, params));
+        return this.extend (this.parseOrder (response), {
+            'status': 'canceled',
+        });
+    }
+
+    async cancelOrders (ids, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let request = { 'orderId': '' };
+        for (let i = 0; i < ids.length; i++) {
+            request += ids[i];
+            if (i < ids.length - 1)
+                request += ',';
+        }
+        let response = await this.v1DeleteOrdersOrderId (this.extend (request, params));
+        return this.extend (this.parseOrder (response), {
+            'status': 'canceled',
+        });
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let response = await this.v1GetBalances (params);
+        let balances = response['data'];
+        return this.parseBalances (balances);
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = 0, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        let request = { 'market': symbol, 'begin': since, 'limit': limit, 'status': 'closed' };
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['market'] = market['id'];
+        }
+        let response = await this.v1GetOrders (this.extend (request, params));
+        let orders = this.parseOrders (response['data'], market, since, limit);
+        return this.filterBySymbol (orders, symbol);
+    }
+
+    async fetchCurrencies (params = {}) {
+        const response = await this.v1GetCoins (params);
+        let currencies = this.parseCurrencies (response['data']);
+        return currencies;
+    }
+
+    async fetchDeposits (code = undefined, since = 0, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        const request = { 'limit': limit, 'page': since };
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['coin'] = currency['id'];
+        }
+        const response = await this.v1GetDeposits (this.extend (request, params));
+        return this.parseDeposits (response['data']);
```

Comment:
This has to fall to the same `parseTransactions` method as with `fetchWithdrawals`.

The exchange usually has this:

```
fetchDeposits → parseTransactions → parseTransaction
fetchWithdrawals → parseTransactions → parseTransaction
```

or this: 

```
fetchTransactions → parseTransactions → parseTransaction
```

The reasons behind that are explained here: https://github.com/ccxt/ccxt/pull/4161#issuecomment-438710176

Reply:
See how this chain is done in Binance, for example.

Reply:
I see, the reason for doing it like that was the difference in fields between both deposits and withdrawals

Reply:
@ahmedbodi yeah, I understand, but we'd rather keep it aligned with the other exchanges, suppose what happens if the user relies on our unified structure and redefines `parseTransaction` for that matter. It will break down (won't work) with this exchange, right?

Reply:
I reckon so

Reply:
Reading the conversation in that commit. The api has a unique endpoint for deposits, withdrawals and transactions. does it still mean they need to run through parseTransactions?

Reply:
Yep, that's right. They all fall into to the same parsing chain.

Reply:
@ahmedbodi one more thing on this: we need to make sure that we don't override the `symbol` or `currency` in `parse*` methods for any of unified structured. If you look at them closer, you will notice that `parse*` methods accept a market or a currency (undefined by default). We **should not** rely on the `market` or `currency` argument being passed from the outside to the `parse*` method and should take the currency code or market pair and deduce it from the structure itself (internally, without using external hints) – as much as possible. The second argument of any `parse*` method is only used with exchanges that don't include all the necessary info to identify the market or currency properly from the response itself. To get this better, see how markets and currencies are parse with respect to this.commonCurrencyCode and unified ids vs symbols in `parse*` methods for hitbtc2, bitfinex, kraken, binance.